flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBHandle.java

```diff
@@ -202,12 +202,21 @@ private void restoreInstanceDirectoryFromPath(Path source) throws IOException {
             final String fileName = file.getFileName().toString();
             final Path targetFile = instanceRocksDBDirectory.resolve(fileName);
             if (fileName.endsWith(SST_FILE_SUFFIX)) {
-                // hardlink'ing the immutable sst-files.
-                Files.createLink(targetFile, file);
-            } else {
-                // true copy for all other files.
-                Files.copy(file, targetFile, StandardCopyOption.REPLACE_EXISTING);
+                try {
+                    // hardlink'ing the immutable sst-files.
+                    Files.createLink(targetFile, file);
+                    continue;
+                } catch (IOException ioe) {
+                    logger.debug(
+                            String.format(
+                                    "Could not hard link sst file %s. Trying to copy it over.",
+                                    fileName),
+                            ioe);
+                }
             }
+
+            // true copy for all other files and files that could not be hard linked.
+            Files.copy(file, targetFile, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Does this move to catch block？ And meanwhile remove the continue.

Reply:
There exist other files in RocksDB folder without `.sst` suffix.

Reply:
Yes, it is because of the other non `.sst` files.